text/0328-polyglot-assert.md

```diff
@@ -0,0 +1,336 @@
+| .NET                  | NuGet       | `Amazon.CDK.Assertions`                       |
+| Go                    | -           | `github.com/aws/aws-cdk-go/awscdk/assertions` |
+
+### Is this feature available in the AWS CDK v2?
+
+This module is available in both the AWS CDK v1 and v2, just like all of our other
+modules.
+
+In CDKv2, this module will be available as part of `aws-cdk-lib`.
+The import statement you will need to use is -
+
+```ts
+import { assertions } from 'aws-cdk-lib';
+```
+
+Read more about `aws-cdk-lib` at
+<https://github.com/aws/aws-cdk/tree/master/packages/aws-cdk-lib#readme>.
+
+### Why are all the APIs suffixed with the `Beta1` suffix?
+
+We have promised that there will be no backwards incompatible or breaking changes
+to our APIs in `aws-cdk-lib` as part of the AWS CDK v2.
+
+However, this module is still under development and is a developer preview release,
+where we are collecting feedback from users and performing necessary improvements.
+
+When an API needs to be modified in backwards incompatible ways, we will create a
+new API with a new suffix (`Beta2`, etc.). The old APIs will continue to be work but
+will be marked as deprecated.
+
+When the module is ready for prime time, we will publish the APIs without the 'Beta'
+suffix and mark all 'Beta' APIs as deprecated.
+
+## Internal FAQ
+
+### Why are we doing this?
+
+We have received a lot of requests and feedback from users that they would like to use
+our 'assert' library in the languages of their choice, primarily Python users.
+
+### Why should we _not_ do this?
+
+None.
+
+### What changes are required to enable this change?
+
+The high level design is to create a new jsii module in the AWS CDK monorepo -
+`@aws-cdk/assertions` - that exposes the APIs proposed above.
+
+The implementation of this module would simply be a scaffold around the existing
+`assert` library.
+
+To build this scaffold, we will 'vendor in' the libraries `assert-internal`,
+`cloudformation-diff` and `cfnspec`.
+
+See [Appendix A](#appendix-a---detailed-design) for the design and rationale.
+
+Prototype / Implementation: <https://github.com/aws/aws-cdk/pull/14952>.
+
+### Is this a breaking change?
+
+This is not a breaking change.
+
+### What are the drawbacks of this solution?
+
+1. Ideally, the dependencies, `assert`, `cloudformation-diff` and `cfnspec`
+   are all available to be consumed as regular dependencies but are not,
+   since they are part of the same monorepo.
+
+   Instead, the design 'vendors in' these dependencies, i.e., copies over
+   the source code and rewrites imports, thereby increasing tech debt.
+
+2. Any new feature or bug fix to the new assert module that require changes
+   to the API needs to be performed both in the old assert module and new.
+   This can lead to weird/complicated implementation, since they both follow
+   different design patterns.
+
+3. The 'jest' friendly interfaces that were previously available are no longer
```

Comment:
How about we also plan to create framework-specific libraries in addition to the core library?

So, we would have `@aws-cdk/assertions-jest`, `@aws-cdk/assertions-nodeunit`, `aws-cdk-assertions-junit`, etc. Each of them would depend on the core assertions library.

Thoughts on this?

Reply:
Answered in the FAQ.